### PR TITLE
JavaScript: Weather v1.1

### PR DESCRIPTION
**Pull request by mke2111**
## Weather v1.1

- In this branch, I implemented ES6 features to design the weather app. 
- A user inputs the city name and the weather data of that given city is returned.
- TailwindCSS is used for styles.

### CODE REVIEW

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work. 💻 👀 ✨ 

Congratulations: the essential functions of your project are quite complete. I tested the city search function and it works. The unit toggle button works as expected as well. All in all, it's a pretty solid project.

## However, there are some changes required 👀 🧰 🔨 🔧 
Let's try to improve our code modulation. I noticed you only use two files to separate your code: `js/index.js` and `js/content.js`. Your `content.js` file contains mostly `DOM` stuff, which is good, but your `index.js` file mixes logic, DOM and API calls.

Please take some time to organize your code into different files, according to:

1. DOM related stuff. All your calls to DOM for creating HTML elements should be contained in a single, separate file.
2. API calls. All the functionality relating to communication with the asynchronous weather API should be contianed in another separate file.
3. APP logic. This is where you bring the previous two modules together to build the complete project. Again: this should be in a separate file.

Also, please take some time to apply the following requirement:

image

Some weather APIs offer image URLs which reflect the weather from the queried city. Please use this, or some change in background color to satisfy this project requirement.

***[ OPTIONAL ]***
- You may add a `Loading` function while the user waits for the API weather data to load.
- You may also automatically detect the user's location on page load, so that their weather data displays without them having to search for their city name.
### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Comment by mke2111** [comment]
### New Changes: ⚪ 

- Separated Logic from the DOM.
- Update background depending on the weather of the city.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @mke2111  👋

**You've done pretty well here fixing a few issues raised in the previous review. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "extends": ["airbnb-base"],
+  "rules": {
+    "no-shadow": "off",
+    "no-param-reassign": "off",
+    "eol-last": "off",
+    "arrow-parens": "off",
+    "no-unused-vars": "off"
```
Comment (mikenath223) [inline .eslintrc.json]: - [x] Kindly remove the no-unused vars rule, as it doesn't correlate to the linting rules shared by other students. Kindly use only the rules set out [here](https://github.com/microverseinc/linters-config/tree/master/javascript)

File: src/js/index.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  };
+};
+
+// Get weather data
+
+async function getData(city) {
+  const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=49257f6591cfc3ed8daf0b5970d519cb`,
+    { mode: 'cors' });
+
+  const data = await response.json();
```
Comment (mikenath223) [inline src/js/index.js]: - [x] Please change this to es6 arrow function syntax. Do this for other regular functions in the code too. It's important to use es6+ rules.

File: .eslintrc.json (hunk 1 above)
Comment (mke2111) [reply .eslintrc.json]: It's funny that I was told to add that line in the previous project and they claimed it was mandatory. I just like the code reviewers. 😆

**Comment by mke2111** [comment]
## New Changes 🆕 

- Change only one async function to match ES6 syntax.
- Update eslintrc.json file

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello again 👋

**You've done pretty well again fixing some issues raised in the previous review. 👌**

However, there is one more fix left. Kindly hang in there.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/js/index.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+// modal
+
+const modalWarn = () => {
+  modal.style.display = 'block';
+
+  window.onclick = function (event) {
+    if (event.target === modal) {
+      modal.style.display = 'none';
+    }
+  };
```
Comment (mikenath223) [inline src/js/index.js]: - [x] Kindly change this to es6 arrow function syntax.

**Comment by mke2111** [comment]
## New Changes 🌑 

- change modal exit fn to match ES6 syntax

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**